### ISSUE TITLE
Enhance the compatibility with other chat-related mods.

### ISSUE THREAD

**Issue by fucksophie** [fuji-fabric/fuji#66]
Setting my prefix to `<red>Some colors for you` in luckperms, and having `%fuji:player_prefix%%player:displayname_visual%%fuji:player_suffix%: %message%` as my format, I get no colors from the prefix. Colors also don't work in the format.

Modlist:
![image](https://github.com/user-attachments/assets/7e2272dc-35b4-4f85-96c8-77ae4e57da1d)

Hover and other things also do not work. 
It works if I leave and join and the `chat.history` submodule is turned on.

**Comment by fucksophie** [comment]
This is a issue with DCIntegration.

If you enable
```
[compatibility]
  # Disables modifying sent chat messages in-game to show who was pinged. (Will not convert <@1234567890> to @User)
  # This may fix mods also modifying sent messages
  disableParsingMentionsIngame = true
  ```
this setting, it fixes itself.

**Comment by sakurawald** [comment]
> Setting my prefix to `<red>Some colors for you` in luckperms, and having `%fuji:player_prefix%%player:displayname_visual%%fuji:player_suffix%: %message%` as my format, I get no colors from the prefix. Colors also don't work in the format.
> 
> Modlist: ![image](https://private-user-images.githubusercontent.com/81387799/363190396-7e2272dc-35b4-4f85-96c8-77ae4e57da1d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUwODAxMjEsIm5iZiI6MTcyNTA3OTgyMSwicGF0aCI6Ii84MTM4Nzc5OS8zNjMxOTAzOTYtN2UyMjcyZGMtMzViNC00Zjg1LTk2YzgtNzdhZTRlNTdkYTFkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODMxVDA0NTAyMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTViMDY5MmIxMmViMGI4OTk1MWQ0ZTFhYzQyN2VjNDU3MWE3ZWUyNzhjY2U1YzExN2QzNjQzOWRkNGRkZWZlMjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Px-04g_MUDaEzfsINdfU8wD__XGXB1gF6XUYDuzwErc)
> 
> Hover and other things also do not work. It works if I leave and join and the `chat.history` submodule is turned on.

This messages sent by `chat.history` are not `chat message`. Acturally, they are buffered messages that only sent to the new-joined player.

**Comment by sakurawald** [comment]
> This is a issue with DCIntegration.
> 
> If you enable
> 
> ```
> [compatibility]
>   # Disables modifying sent chat messages in-game to show who was pinged. (Will not convert <@1234567890> to @User)
>   # This may fix mods also modifying sent messages
>   disableParsingMentionsIngame = true
> ```
> 
> this setting, it fixes itself.

This seems to be an issue on chat message processing event priority. Maybe fuji should process the chat message later, so that the time-sequence is expected.

**Comment by sakurawald** [comment]
In current stage, fuji listens on the vanilla method "handleDecoratedMessage()" with 1001 priority (the default priority + 1). Some other chat-related mods may listen on other methods. Since fabric platform doesn't provide a chat event like bukkit api, it's hard to decide the priority of processing chat-message.

**Comment by sakurawald** [comment]
This should be fixed in `fuji v1.6.3`.
